backend/Functions/Edna.Connect/LtiAdvantageApi.cs

```diff
@@ -106,6 +106,9 @@ public async Task SaveState([OrchestrationTrigger] IDurableOrchestrationContext
             await assignmentsCollector.AddAsync(assignment);
             await assignmentsCollector.FlushAsync();
 
+            if (string.IsNullOrEmpty(assignment.Id))

```

Comment:
Amazing. Let's add a descriptive message / log to indicate that the failure is due to missing assignment id.

Reply:
assignment Id is not created from context id and resource link id when the assignment doesn't save successfully, an error message with the reason for failure does get logged.

Reply:
Again, I am unable to understand your answer. From where does the logging happens in the above case? Are we seeing this in the logs where this is happening? Request you to please point me to the location to help me get the necessary context.

Reply:
Assignment.Id is gets updated in the SaveAssignment function (which is called from the flushAsync function) when the assignment gets saved successfully.
In case the assignment save is not successful, assignment.Id is not updated (remains null) and an error msg with appropriate reason gets logged in the LtiAdvantage/ Lti1 function.
The logging happens from the SaveAssignment function in the `AssigmentsClient.cs` file.

I hope this clears your doubt.

Reply:
The same check is present in the Lti1Api.cs file as well.